productcomposer: add ignore_binaries_newer_than pkgset option

g7 opened the pull request:
It allows to ignore binaries with a buildtime newer than
specified. A valid datetime as parsed by pydantic can
be used.
    
        ignore_binaries_newer_than: 2025-12-30 00:00
    
This field is optional. When not taking all the available
binary versions, the closest package with a valid buildtime
is used instead.
    
Note that newly added packages (that do not have an older
version to fallback on) will be skipped as well, and it
might abort the build if the `ignore_missing_packages` is
not enabled.

adrianschroeter reviewed (changes requested):
In general, I wonder if we could not simplify this a lot by just ignoring rpms build newer as a given time.

Updateinfo.xml would be skipped then anyway and your product would not have updateinfo.xml at all enabled, right?

IMHO this would simplify the logic and code a lot.

Would this work for you?

g7 committed:
productcomposer: add ignore_binaries_newer_than pkgset option

It allows to ignore binaries with a buildtime newer than
specified. A valid datetime as parsed by pydantic can
be used.

    ignore_binaries_newer_than: 2025-12-30 00:00

This field is optional. When not taking all the available
binary versions, the closest package with a valid buildtime
is used instead.

Note that newly added packages (that do not have an older
version to fallback on) will be skipped as well, and it
might abort the build if the `ignore_missing_packages` is
not enabled.

Signed-off-by: Eugenio Paolantonio <eugenio.paolantonio@suse.com>

g7 committed:
productcomposer: createupdateinfoxml: set rpm buildtime by looking at the issued date

If we're skipping packages using a DateFilter, when processing
updateinfos we might die on looking for rpms that were filtered out.

Set the updateinfo issued date as the buildtime in the fake package
entry used for matching. Although it is not precise, it should help in
properly skipping the referenced packages.

Signed-off-by: Eugenio Paolantonio <eugenio.paolantonio@suse.com>
